src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.cpp

```diff
@@ -181,6 +164,63 @@ ep_rt_aot_diagnostics_command_line_get (void)
 #endif
 }
 
+namespace
+{
+    #ifdef TARGET_UNIX
+    __thread EventPipeThreadHolder* eventpipe_tls_instance;
+    #else
+    thread_local EventPipeThreadHolder* eventpipe_tls_instance;
+    #endif
+
+    void free_thread_holder ()
+    {
+        EventPipeThreadHolder *thread_holder = eventpipe_tls_instance;
+        if (thread_holder != NULL) {
+            thread_holder_free_func (thread_holder);
+            eventpipe_tls_instance = NULL;
+        }
+    }
+
+    EventPipeThreadHolder* get_thread_holder ()
+    {
+        return eventpipe_tls_instance;
+    }
+
+    EventPipeThreadHolder* create_thread_holder ()
+    {
+        STATIC_CONTRACT_NOTHROW;
+
+        // Free any existing holder
```

Comment:
When do we end up with existing holder that needs to be freed?

Reply:
Never, it would appear. I stuck to the existing logic a bit too much - will remove.